src/main/java/org/jitsi/jicofo/AbstractChannelAllocator.java

```diff
@@ -0,0 +1,394 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo;
+
+import net.java.sip.communicator.impl.protocol.jabber.extensions.colibri.*;
+import net.java.sip.communicator.impl.protocol.jabber.extensions.jingle.*;
+import net.java.sip.communicator.impl.protocol.jabber.extensions.jitsimeet.*;
+import net.java.sip.communicator.impl.protocol.jabber.jinglesdp.*;
+import net.java.sip.communicator.service.protocol.*;
+import org.jitsi.jicofo.discovery.*;
+import org.jitsi.jicofo.util.*;
+import org.jitsi.protocol.xmpp.*;
+import org.jitsi.protocol.xmpp.util.*;
+import org.jitsi.util.*;
+import org.jxmpp.jid.*;
+
+import java.util.*;
+
+/**
+ * A thread which allocates Colibri channels and (optionally) initiates a
```

Comment:
i think it would be better to separate the logic for allocating the channels and initiating a session...keep classes to a single responsibility.  was there a reason to have this do both?

Reply:
It was just that the allocation and invite happens on separate thread which is basically managed by this class. I don't really see the point in having another class just to send the invite. Jingle stuff is handled by separate thing anyway.